### DISCUSSION TITLE
Scanorama error: passed valued is not matched to AnnData shape

### DISCUSSION THREAD

**Discussion by MikeZhao03** [brianhie/scanorama#150] [Q&A]
Hello everyone, 

I am pretty new to python, and I am trying to use scanorama to integrate my two samples. 
I just follow the tutorial of scanpy and had this error, it seems the numbers doesn't match but I don't how to fix it. Any advice will be greatly appreciated! thanks!

ValueError: Length of passed value for var_names is 19465, but this AnnData has shape: (1526, 19454)

**Comment by fabricio-fn** [comment]
The error states that the length of variable names is different between the passed value and the AnnData format.

it is important to ensure that the variable names are correctly aligned.

check the variable names in each sample using the AnnData object's var_names attribute.

print(adata_A.var_names)
print(adata_B.var_names)

**Reply by MikeZhao03** [reply]
Hi, thanks for you reply. 
I just check the code, the var_names seems are same 
my two samples have both 19465 var. 
When I run adatas_cor = scanorama.correct_scanpy(adatas) it become 19454 genes. 

Found 19454 genes among all datasets
[[0.         0.74639581]
 [0.         0.        ]]
Processing datasets (0, 1)

**Comment by fabricio-fn** [comment]
To resolve this issue, you need to ensure that the number of variable names provided matches the number of variables (resources) in your AnnData object.

**Comment by fabricio-fn** [comment]
Check if there are any parameters in the correct_scanpy function that affect the gene count. Some parameters can control how genes are treated during correction.

also check that the input data "AnnData objects" is correctly formatted and contains the genes in the expected order. Make sure both samples have the same set of genes (19465) before correction.
